common/persistence/persistence_rate_limited_clients.go

```diff
@@ -45,50 +45,66 @@ const (
 
 var (
 	// ErrPersistenceLimitExceeded is the error indicating QPS limit reached.
-	ErrPersistenceLimitExceeded = serviceerror.NewResourceExhausted(enumspb.RESOURCE_EXHAUSTED_CAUSE_PERSISTENCE_LIMIT, "Persistence Max QPS Reached.")
+	ErrPersistenceLimitExceeded = serviceerror.NewResourceExhausted(
+		enumspb.RESOURCE_EXHAUSTED_CAUSE_PERSISTENCE_LIMIT,
+		enumspb.RESOURCE_SCOPE_SYSTEM,
+		"Persistence System Max QPS Reached.",
```

Comment:
```suggestion
		"System Persistence Max QPS Reached.",
```

Scope then resource reads a bit better